Automated Smoketest 
Sha: e30894de57b02fa6ea435e2b545c998be871aa95
Build: failed
Number of Error(s): 4
Error: no ID for constraint linkend: ssl4esp.
/mnt/disk1/home/vamosax/smoketest/PR-9095/HPCC-Platform/esp/bindings/SOAP/soaplib/./../../../../system/security/shared/../../../system/security/shared/secloader.hpp:70:48: error: \xe2\x80\x98bnd_cfg\xe2\x80\x99 was not declared in this scope
/mnt/disk1/home/vamosax/smoketest/PR-9095/HPCC-Platform/esp/bindings/SOAP/soaplib/./../../../../system/security/shared/../../../system/security/shared/secloader.hpp:71:5: error: control reaches end of non-void function [-Werror=return-type]
cc1plus: some warnings being treated as errors

--- a/system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp
+++ b/system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp
@@ -32,20 +32,24 @@ public:
         if(pwFile.isEmpty())
             throw MakeStringException(-1, "htpasswdFile not found in configuration");
 
-        Owned<IPropertyTree> authcfg = bindConfig->getPropTree("Authenticate");
-        if(authcfg != nullptr)
         {
-            StringBuffer authxml;
-            toXML(authcfg, authxml);
-            DBGLOG("HTPASS Authenticate Config: %s", authxml.str());
+            Owned<IPropertyTree> authcfg = bindConfig->getPropTree("Authenticate");
+            if(authcfg != nullptr)
+            {
+                StringBuffer authxml;
+                toXML(authcfg, authxml);
+                DBGLOG("HTPASS Authenticate Config: %s", authxml.str());
+            }
         }
 
-        Owned<IPropertyTree> custombindingconfig = bindConfig->getPropTree("CustomBindingParameters");
-        if(custombindingconfig != nullptr)
         {
-            StringBuffer custconfigxml;
-            toXML(custombindingconfig, custconfigxml);
-            DBGLOG("HTPASS Custom Binding Config: %s", custconfigxml.str());
+            Owned<IPropertyTree> custombindingconfig = bindConfig->getPropTree("CustomBindingParameters");
+            if(custombindingconfig != nullptr)
+            {
+                StringBuffer custconfigxml;
+                toXML(custombindingconfig, custconfigxml);
+                DBGLOG("HTPASS Custom Binding Config: %s", custconfigxml.str());
+            }
         }
 
         apr_initialized = false;

--- a/system/security/shared/secloader.hpp
+++ b/system/security/shared/secloader.hpp
@@ -35,12 +35,12 @@ public:
     /// for the given ESP service
     ///
     /// @param  bindingName     Binding name ie 'WsTopology_smc_myesp'
-    /// @param  bnd_cfg         'Binding' IPropertyTree associated with ESPService
+    /// @param  bindingCfg      'Binding' IPropertyTree associated with ESPService
     /// @param  secMgrCfg       'SecurityManager' IPropertyTree from component config file
     ///
     /// @return an ISecManager Security Manager instance
     ///
-    static ISecManager* loadPluggableSecManager(const char * bindingName, IPropertyTree* bnd_cfg, IPropertyTree* secMgrCfg)
+    static ISecManager* loadPluggableSecManager(const char * bindingName, IPropertyTree* bindingCfg, IPropertyTree* secMgrCfg)
     {
         const char * lsm = "Load Security Manager :";
 
@@ -67,7 +67,7 @@ public:
 
         //Call ISecManager instance factory and return the new instance
         DBGLOG("Calling '%s' in pluggable security manager '%s'", instFactory.str(), libName.str());
-        return xproc(bindingName, *secMgrCfg, *bnd_cfg);
+        return xproc(bindingName, *secMgrCfg, *bindingCfg);
     }
 
     static ISecManager* loadSecManager(const char* model_name, const char* servicename, IPropertyTree* cfg)

